pkg/alert/alert.go

```diff
@@ -294,21 +310,27 @@ func (s *Sender) Send(ctx context.Context, alerts []*Alert) error {
 			if err := s.sendOne(sendCtx, amURL.String(), b); err != nil {
 				level.Warn(s.logger).Log(
 					"msg", "sending alerts failed",
-					"alertmanager", u.Host,
-					"numDropped", len(alerts),
+					"alertmanager", amURL.Host,
+					"numAlerts", len(alerts),
 					"err", err)
-				s.dropped.WithLabelValues(u.Host).Add(float64(len(alerts)))
-				return err
+				s.errs.WithLabelValues(amURL.Host).Inc()
+				return
+			} else {
+				atomic.AddUint64(&numSuccess, 1)
 			}
-			s.sent.WithLabelValues(u.Host).Add(float64(len(alerts)))
-			s.latency.WithLabelValues(u.Host).Observe(time.Since(start).Seconds())
-			return nil
-		})
+			s.sent.WithLabelValues(amURL.Host).Add(float64(len(alerts)))
+			s.latency.WithLabelValues(amURL.Host).Observe(time.Since(start).Seconds())
+		}()
 	}
-	if err := g.Wait(); err != nil {
-		return errors.Wrap(err, "send alerts")
+	wg.Wait()
+
+	if numSuccess > 0 {
```

Comment:
I assume sending alerts to AM is an all or nothing thing. Either it successfully acks all alerts or none at all.

Reply:
Because that is the case. ^^

That's why alertmanager HA exists. If we have 3 and we cannot contact 1 of them, we still succeded.

We send all alerts everywhere. (: 

This is 1:1 implementation of